include/utf8string.h

```diff
@@ -0,0 +1,207 @@
+#ifndef NEWSBOAT_UTF8STRING_H_
+#define NEWSBOAT_UTF8STRING_H_
+
+#include <functional>
+#include <string>
+
+namespace rust {
+inline namespace cxxbridge1 {
+class String;
+class Str;
+}
+}
+
+namespace newsboat {
+
+/// A string that's guaranteed to contain valid UTF-8.
+class Utf8String {
+public:
+	Utf8String() = default;
+
+	// XXX: Hack for the duration of converting.
+	// These should reduce some of the churn, when working
+	// between converted and old interfaces.
+	Utf8String(std::string input);
+	operator std::string() const
+	{
+		return inner;
+	}
```

Comment:
I was considering this, and there definitely is some risk in having these.

I still think the transition will be considerably smoother with these.
Since all newsboat code will eventually use `Utf8String`, explicit conversions would introduce a lot more churn for changing a single class/function.

With explicit conversions, changing one function would require adding/removing explicit conversions in:
- all uses of the function
- all calls to other newsboat functions in the function body

This would basically mean a lot of diff pairs like:

```patch
- f(s);
+ f(s.utf8());
```

```patch
- f(s.utf8());
+ f(s);
```


Overall, I think the number of calls between newsboat code makes the implicit conversions worth the risk.